tests/Microsoft.Identity.Web.Test/WebAppExtensionsTests.cs

```diff
@@ -1,32 +1,493 @@
+                    options.Events.OnTokenValidated += tokenValidatedFuncMock;
+                    options.Events.OnRedirectToIdentityProviderForSignOut += redirectFuncMock;
+                });
+
+            services.RemoveAll<ITokenAcquisition>();
+            services.AddScoped<ITokenAcquisition>((provider) => tokenAcquisitionMock);
+
+            var provider = services.BuildServiceProvider();
+
+            // Assert configure options actions added correctly
+            var configuredAppOptions = provider.GetServices<IConfigureOptions<ConfidentialClientApplicationOptions>>().Cast<ConfigureNamedOptions<ConfidentialClientApplicationOptions>>();
+            var configuredMsOptions = provider.GetServices<IConfigureOptions<MicrosoftIdentityOptions>>().Cast<ConfigureNamedOptions<MicrosoftIdentityOptions>>();
+
+            Assert.Contains(configuredAppOptions, o => o.Action == _configureAppOptions);
+            Assert.Contains(configuredMsOptions, o => o.Action == _configureMsOptions);
+
+            var oidcOptions = provider.GetRequiredService<IOptionsFactory<OpenIdConnectOptions>>().Create(_oidcScheme);
+
+            AddWebAppCallsProtectedWebApi_TestCommon(services, provider);
+            await AddWebAppCallsProtectedWebApi_TestAuthorizationCodeReceivedEvent(provider, oidcOptions, authCodeReceivedFuncMock, tokenAcquisitionMock).ConfigureAwait(false);
+            await AddWebAppCallsProtectedWebApi_TestTokenValidatedEvent(oidcOptions, tokenValidatedFuncMock).ConfigureAwait(false);
+            await AddWebAppCallsProtectedWebApi_TestRedirectToIdentityProviderForSignOutEvent(provider, oidcOptions, redirectFuncMock, tokenAcquisitionMock).ConfigureAwait(false);
+        }
+
+        private void AddWebAppCallsProtectedWebApi_TestCommon(IServiceCollection services, ServiceProvider provider)
+        {
+            // Assert correct services added
+            Assert.Contains(services, s => s.ServiceType == typeof(IHttpContextAccessor));
+            Assert.Contains(services, s => s.ServiceType == typeof(ITokenAcquisition));
+            Assert.Contains(services, s => s.ServiceType == typeof(IConfigureOptions<ConfidentialClientApplicationOptions>));
+            Assert.Contains(services, s => s.ServiceType == typeof(IConfigureOptions<MicrosoftIdentityOptions>));
+            Assert.Contains(services, s => s.ServiceType == typeof(IConfigureOptions<OpenIdConnectOptions>));
+
+            // Assert OIDC options added correctly
+            var configuredOidcOptions = provider.GetService<IConfigureOptions<OpenIdConnectOptions>>() as ConfigureNamedOptions<OpenIdConnectOptions>;
+
+            Assert.Equal(_oidcScheme, configuredOidcOptions.Name);
+        }
+
+        private async Task AddWebAppCallsProtectedWebApi_TestAuthorizationCodeReceivedEvent(IServiceProvider provider, OpenIdConnectOptions oidcOptions, Func<AuthorizationCodeReceivedContext, Task> authCodeReceivedFuncMock, ITokenAcquisition tokenAcquisitionMock)
```

Comment:
nit: can you move each parameter to a new line for the longer method signatures? thx.

Reply:
Done. Do we follow a parameter limit or just when it wraps around or arbitrary?

Reply:
we don't have a specific limit. @bgavrilMS started this for us in .NET. Usually when we have 3+, we start making new lines.

---
In reply to: [417819635](https://github.com/AzureAD/microsoft-identity-web/pull/132#discussion_r417819635) [](ancestors = 417819635)